strings/base_error.h

```diff
@@ -546,6 +546,28 @@ namespace winrt::impl
         }
         return result;
     }
+
+    template <typename T>
+    WINRT_IMPL_NOINLINE void check_cast_result(T* from WINRT_IMPL_SOURCE_LOCATION_ARGS)
```

Comment:
Doesn't need to be a template. Can just take a `void*`. This allows the different instantiations to be folded.

Reply:
That's fair.  If I spin another PR I'll make that change.